### PR TITLE
Update seq_region.coord_system_id

### PR DESCRIPTION
**Pull request by helensch**
PR for ENSVAR-3007
Updates the seq_region.coord_system_id and populates the coord_system table so that Foreign Key DataCheck passes

### CODE REVIEW

File: scripts/misc/update_seq_region_coord.pl [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::DBSQL::DBConnection;
+use Getopt::Long;
```
Comment (ima23) [inline scripts/misc/update_seq_region_coord.pl:21]: Is this library required or is it redundant when using the CliHelper?

File: scripts/misc/update_seq_region_coord.pl [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+use Bio::EnsEMBL::Utils::CliHelper;
+
+my $script_opts = [{args => ['host', 'dbhost', 'h'], type => '=s'},
```
Comment (ima23) [inline scripts/misc/update_seq_region_coord.pl:24]: Can `script_opts` be formatted in the expected `optsd` at initialisation and skipping the need of the `get_opts` method which formats one variable into the other?

File: scripts/misc/update_seq_region_coord.pl [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      print "$variation_dbname: No foreign key errors between seq_region and coord_system tables\n";
+  } else {
+    print "$variation_dbname: $fk_count foreign key errors between seq_region and coord_system tables\n";
+  }
+}
+
+sub get_dbas { 
+  my $opts = shift;
+  my $dbas = {};
+  for my $db_args ( @{ $cli_helper->get_dba_args_for_opts( $opts, 0 ) } ) {
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:129]: A comment on cli call has been added as in synopsis in CliHelper docs.
The error message is from CliHelper code and does not include the dbname.
It occurs when species.production_name not specified in meta. The issue has been added to ENSVAR-3254

File: scripts/misc/update_seq_region_coord.pl [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+my $cli_helper = Bio::EnsEMBL::Utils::CliHelper->new();
+my $optsd = get_opts();
+
+my $opts = $cli_helper->process_args( $optsd, \&usage );
+
+my ($variation_dbas, $core_dbas);
+if (defined $opts->{dbname}) {
+  $variation_dbas = get_dbas($opts);
+  die "Variation database required" unless ($opts->{dbname} =~ m/variation/);
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:44]: Change has been made. Added to ENSVAR-3253 for update_meta_coord.pl

File: scripts/misc/update_seq_region_coord.pl [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+  $0 --dbhost=XhostX [--dbport=3306] \\
+  \t--dbuser=XuserX --dbpass=XXX \\
+  $0 \$(XserverX details script_db) --dbname XdbnameX
+  $0 \$(XserverX details script_db) --dbname XdbnameX --dry_run
+  $0 \$(XserverX details script_db) --version Xrelease-versionX
+  $0 --help
+  --dry_run     Print update statements
+  --help        Displays this help text.
+
+This script will update
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:158]: Added to the help

File: scripts/misc/update_seq_region_coord.pl [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  my $version = $opts->{version};
+  die "Version argument is required" unless ($version);
+  $opts->{pattern} = 'variation_'.$version.'_\d+$';
+  $variation_dbas = get_dbas($opts);
+}
+
+foreach my $variation_dbname (keys %$variation_dbas) {
+  print "variation db: $variation_dbname\n";
+  my $core_dbname = $variation_dbname;
+  $core_dbname =~ s/variation/core/;
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:56]: Issue added to ENSVAR-3254

File: scripts/misc/update_seq_region_coord.pl [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        {args => ['version'],    type => ':i'},
+        {args => ['dry_run'], type=>'!'}, ];
+
+if (scalar(@ARGV) == 0) {
+  usage();
+  exit 0;
+}
+
+my $cli_helper = Bio::EnsEMBL::Utils::CliHelper->new();
+my $optsd = get_opts();
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:38]: From update_meta_coord.pl. Issue added to ENSVAR-3254, ENSVAR-3253

File: scripts/misc/update_seq_region_coord.pl [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  my $core_dbname = $variation_dbname;
+  $core_dbname =~ s/variation/core/;
+  print "core db: $core_dbname\n";
+  my $vdba = $variation_dbas->{$variation_dbname};
+  next unless ($vdba);
+  
+  # Are there foreign key failures between seq_region and coord_system?
+  my $fk_count = count_fk_failures($vdba);
+  if ($fk_count == 0) {
+    print "$variation_dbname: No foreign key errors between seq_region and coord_system tables\n";
```
[the remark this answers is not in the document]
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:64]: The use case for the dbname is for tracking and debugging.

File: scripts/misc/update_seq_region_coord.pl (hunk 1 above)
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:21]: From update_meta_coord.pl. Issue added to ENSVAR-3254, ENSVAR-3253

File: scripts/misc/update_seq_region_coord.pl (hunk 2 above)
Comment (helensch) [reply scripts/misc/update_seq_region_coord.pl:24]: From update_meta_coord.pl. Issue added to ENSVAR-3254, ENSVAR-3253

**Comment by helensch** [comment]
@ima23 Thank you for the review. Minor fixes done. Other issues added to ENSVAR-3254 and ENSVAR-3253 (update_meta_coord.pl)

**Comment by ima23** [comment]
merged to `release/101` and `master`

**Pull request merged**